tool/internal_pkg/generator/custom_template.go

```diff
@@ -199,6 +199,79 @@ func (g *generator) GenerateCustomPackage(pkg *PackageInfo) (fs []*File, err err
 	return fs, nil
 }
 
+func readTpls(rootDir, currentDir string, ts []*Template) ([]*Template, error) {
+	files, _ := os.ReadDir(currentDir)
+	for _, f := range files {
+		// filter dir and non-tpl files
+		if f.IsDir() {
+			subDir := filepath.Join(currentDir, f.Name())
+			subTemplates, err := readTpls(rootDir, subDir, ts)
+			if err != nil {
+				return nil, err
+			}
+			ts = append(ts, subTemplates...)
+		} else if strings.HasSuffix(f.Name(), ".tpl") {
```

Comment:
What if there is a .tpl file in the user's directory and we do not need to render it?